.github/scripts/filter_test_configs.py

```diff
@@ -558,6 +559,12 @@ def main() -> None:
             args.workflow, args.job_name, filtered_test_matrix
         )
 
+    if MEM_LEAK_LABEL in labels:
```

Comment:
Nits: I think you could move this into `perform_misc_tasks` as the function's name implies